ev3dev/sensor/lego.py

```diff
@@ -430,6 +430,14 @@ def reset(self):
         self.mode = self.MODE_GYRO_ANG
         self._direct = self.set_attr_raw(self._direct, 'direct', 17)
 
+    def wait_until_changed_by(self, delta):
+        """
+        Wait until angle has changed by specified amount.
+        """
+        start_angle = self.value()
```

Comment:
- would call it something like `wait_until_angle_changed_by()`
- have it assert the mode is one of MODE_GYRO_ANG, MODE_GYRO_G_A, or MODE_TILT_ANG
- if it is MODE_GYRO_G_A you'll want to use `self.value(1)`, for the other two `self.value(0)`
- you could also do a sibling `wait_until_rate_changed_by()` method...or does that make any sense? Maybe `wait_until_rate()` instead?

Reply:
> you could also do a sibling wait_until_rate_changed_by() method...or does that make any sense? Maybe wait_until_rate() instead?

Yeah... that makes no sense 😀

Reply:
Thanks, sounds great. I'll change the name and add the assert code. I think the equivalent EV3-G block has functionality similar to wait_until_rate_changed_by() and wait_until_rate(). With wait_until_rate(), I need to have a think about how that should work. In EV3-G, you have to specify a comparison operator like >=, <= etc. 

I'll have a good look at the EV3-G wait block functionality. I'm assuming that the goal of the high level API is to mirror how EV3-G works as closely as possible. 

Would it be okay to address wait_until_rate_changed_by() and wait_until_rate() in a separate pull request ? Or is it better to include it in this one ?

Reply:
> Would it be okay to address wait_until_rate_changed_by() and wait_until_rate() in a separate pull request

yep that would be fine

Reply:
Looks like MODE_GYRO_G_A actually also uses self.value(0). The rate seems to be at self.value(1).

The rate_and_angle property might need to be renamed angle_and_rate or the order of the return statement reversed.

Reply:
> would call it something like wait_until_angle_changed_by()

> you could also do a sibling wait_until_rate_changed_by() method...

We have `Motor.wait()` with generic condition. Would it make sense to add some common predicates like `angle_changed_by(n)`, `rate_changed_by(n)`? To be used like
```py
m.wait(m.angle_changed_by(n))
```
The predicates could be implemented as
```py
def angle_changed_by(self, n):
    cur_angle = self.angle
    def changed_by():
      return abs(self.angle - cur_angle) >= n
```

Reply:
I've updated the pull request with the new method name and assert code. Tested okay on my robot. I could not test MODE_TILT_ANG as my robot's gyro sensor does not support that. MODE_GYRO_ANG and MODE_GYRO_G_A worked fine.

Reply:
>We have Motor.wait() with generic condition. Would it make sense to add some common predicates like angle_changed_by(n), rate_changed_by(n)? 

I'm thinking of wait_until_angle_changed_by() as not just being associated with motors. For example, it could be used to detect when the robot's angle has been changed by some external event. Also I think it works very similarly to the equivalent EV3-G block and would be easy to understand for users familiar with EV3-G.

Reply:
> wait_until_angle_changed_by() as not just being associated with motors.

Yes, please ignore my comment. Was made before the first coffee cup in the morning.